README.md

```diff
@@ -4,67 +4,69 @@
 
-The `node-version` input is optional. If not supplied, the node version that is PATH will be used. However, this action will still register problem matchers and support auth features. So setting up the node environment is still a valid scenario without downloading and caching versions.
+The action will first check the local cache for a semver match. If unable to find a specific version in the cache, the action will attempt to download a version of Node.js. It will pull LTS versions from [node-versions releases](https://github.com/actions/node-versions/releases) and on miss or failure will fall back to the previous behavior of downloading directly from [node dist](https://nodejs.org/dist/).
 
-# Usage
+For information regarding locally cached versions of Node.js on GitHub hosted runners, check out [GitHub Actions Virtual Environments](https://github.com/actions/virtual-environments).
 
-See [action.yml](action.yml)
+### Caching packages dependencies
+
+The action has a built-in functionality for caching and restoring npm/yarn dependencies. Supported package managers are `npm`, `yarn`. The `cache` input is optional, and caching is turned off by default.
 
-Basic:
 ```yaml
 steps:
 - uses: actions/checkout@v2
 - uses: actions/setup-node@v2
   with:
     node-version: '14'
+    cache: 'npm'
 - run: npm install
 - run: npm test
 ```
 
-Check latest version:
-
-In the basic example above, the `check-latest` flag defaults to `false`. When set to `false`, the action tries to first resolve a version of node from the local cache. For information regarding locally cached versions of Node on GitHub hosted runners, check out [GitHub Actions Virtual Environments](https://github.com/actions/virtual-environments). The local version of Node in cache gets updated every couple of weeks. If unable to find a specific version in the cache, the action will then attempt to download a version of Node. Use the default or set `check-latest` to `false` if you prefer stability and if you want to ensure a specific version of Node is always used.
-
-If `check-latest` is set to `true`, the action first checks if the cached version is the latest one. If the locally cached version is not the most up-to-date, a version of Node will then be downloaded. Set `check-latest` to `true` it you want the most up-to-date version of Node to always be used.
-
-> Setting `check-latest` to `true` has performance implications as downloading versions of Node is slower than using cached versions
-
+**Caching yarn dependencies:**
 ```yaml
 steps:
 - uses: actions/checkout@v2
 - uses: actions/setup-node@v2
   with:
     node-version: '14'
-    check-latest: true
+    cache: 'yarn'
 - run: npm install
 - run: npm test
 ```
+Yarn caching handles both yarn versions: 1 or 2. 
 
-Matrix Testing:
+> At the moment, only `lock` files in the project root are supported.
```

Comment:
> At the moment, only `lock` files in the project root are supported.

Does this mean that `working-directory` will be ignored? That would be a deal breaker for mixed language mono repos.

Reply:
afaik, actions/setup-node doesn't support `working-directory` argument for now. `working-directory` can't be applied to actions steps (`uses:`).

In the first iteration, we have decided to keep things as simple as possible and we are looking for `package-lock.json` / `yarn.lock` only in repository root. We understand that it won't work for mono-repos (or any complex repos) so our recommendation is continue using [actions/cache](https://github.com/actions/cache/blob/main/examples.md#node---npm) directly for such use-cases for now.

We definitely will keep our eyes on feature requests to add supporting mono-repos but we wouldn't like to include it to the first iteration.